File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        i, j = 0, len(heights) - 1
        max_area = 0

        while i < j:
            width = j - i
            length = min(heights[i], heights[j])
            curr_area = width * length

            if heights[i] < heights[j]:
                i += 1
            else:
                j -= 1
            
            max_area = max(max_area, curr_area)

        return max_area

File: Data Structures & Algorithms/insert-new-interval/submission-1.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        if (intervals.empty()) return {{newInterval[0], newInterval[1]}};

        vector<vector<int>> result;
        int start = newInterval[0], end = newInterval[1];
        int n = intervals.size();

        for (int i = 0; i < n; ++i) {
            if (intervals[i][1] < start) {
                result.push_back(intervals[i]);
            }
            else if (intervals[i][0] > end) {
                result.push_back({start, end});
                result.insert(result.end(), intervals.begin() + i, intervals.end());
                return result;
            }
            else {
                start = min(start, intervals[i][0]);
                end = max(end, intervals[i][1]);
            }
        }

        result.push_back({start, end});
        return result;
    }
};
